python/social_bot/gazebo_agent.py

```diff
@@ -170,6 +190,25 @@ def get_observation(self, teacher, sentence_raw="hello"):
             obs = self.get_camera_observation()
         else:  # observation is pure low-dimentional states
             obs = teacher.get_task_specific_observation(self)
+            if self._goal_conditioned:
+                flat_obs = obs
```

Comment:
maybe move into a new function for this part

Reply:
done.